### PR TITLE
Added ability to update entity's fields. Added new entity and test (w…

### PR DESCRIPTION
**Pull request by shashinadya**
…ill remove it, if PR will be approved).

### CODE REVIEW

File: src/main/java/database/service/JsonDatabaseService.java [hunk 1]
```diff
@@ -245,8 +246,9 @@ String getDatabasePath(Class<? extends BaseEntity> entityClass) {
     }
 
     private <T extends BaseEntity> void updateEntityFields(T outcomeEntity, T incomeEntity) {
```
Comment (mishashin) [inline src/main/java/database/service/JsonDatabaseService.java:252]: I think the same changes should be pplied for `getByFilters` method. Now it searchs only in class fields, but not in supercalsses. Please check and fix if needed. And don't forget to provide full list of fields to `Validator.validateDatabaseFilters` in `getByFilters`.

File: src/main/java/database/service/JsonDatabaseService.java [hunk 2]
```diff
@@ -291,20 +302,30 @@ private <T extends BaseEntity> void saveEntitiesToDatabase(List<T> entities, Pat
     }
 
     private String getGetterName(Class<?> entityClass, String fieldName) {
```
Comment (mishashin) [inline src/main/java/database/service/JsonDatabaseService.java]: I think now we can change `fieldName` from `String` to `Field` type.

File: src/test/java/database/service/JsonDatabaseServiceTest.java [hunk 3]
```diff
@@ -106,6 +107,18 @@ void updateRecordInTableWIthCorrectIdTest() {
         assertEquals(secondStudent, jsonDatabaseService.getById(Student.class, 1));
     }
 
+    @Test
+    void updateOxfordStudent() {
+        jsonDatabaseService.createTable(OxfordStudent.class);
+        OxfordStudent os = new OxfordStudent("N", 4.5, 20);
+        OxfordStudent os2 = new OxfordStudent("M", 4.2, 21);
+
+        jsonDatabaseService.addNewRecordToTable(os);
```
Comment (mishashin) [inline src/test/java/database/service/JsonDatabaseServiceTest.java:116]: Please check that student created and has id=0.

Comment (shashinadya) [reply src/test/java/database/service/JsonDatabaseServiceTest.java:116]: Discussed offline. Decided to do not add extra check.

File: src/main/java/database/service/JsonDatabaseService.java [hunk 4]
```diff
@@ -290,14 +305,13 @@ private <T extends BaseEntity> void saveEntitiesToDatabase(List<T> entities, Pat
...
-    private String getGetterName(Class<?> entityClass, String fieldName) {
-        Field field;
-        try {
-            field = entityClass.getDeclaredField(fieldName);
-        } catch (NoSuchFieldException e) {
-            LOG.error("Unable to get declared field {}", fieldName);
-            throw new RuntimeException(e);
+    private String getGetterName(Field field) {
+        if (field == null) {
+            LOG.error("Unable to find field {} in the class hierarchy", field);
```
Comment (mishashin) [inline src/main/java/database/service/JsonDatabaseService.java]: Please change error message. Something like **Unknown entity field**.

**Review by mishashin — APPROVED** [review]
Approved.

**Pull request merged**